src/coreclr/tools/Common/TypeSystem/Interop/UnmanagedCallingConventions.cs

```diff
@@ -135,6 +135,27 @@ public static UnmanagedCallingConventions GetPInvokeMethodCallingConventions(thi
             return result;
         }
 
+        public static UnmanagedCallingConventions GetDelegateCallingConventions(this TypeDesc delegateType)
+        {
+            Debug.Assert(delegateType.IsDelegate);
+
+            if (delegateType is not EcmaType ecmaDelegate)
```

Comment:
Can we flip this invariant and place the `if (unmanagedCallConv != MethodSignatureFlags.None)` check within the branch? We're calling the same "default" return function twice and it is a bit confusing as to why. I think reducing the calls to one would be clearer.